core/src/main/java/org/apache/james/core/quota/QuotaCount.java

```diff
@@ -48,6 +48,11 @@ public long asLong() {
         return value.orElseThrow(IllegalStateException::new);
     }
 
+    @Override
+    public Optional<Long> getValue() {
```

Comment:
this commit seems to be a bad idea. Value being an optional is an internal representation of the object, but it has no reason to appear in an API.

Reply:
Reason I exposed this API is: I got a trouble with scala implementing, my first impl was:
```java
  implicit val quotaValueWrites: Writes[QuotaValue[_]] = value => {
    if (value.isLimited) {
      JsNumber(value.asLong())  // Even value.isLimited() = true, then this lambda call still returns JsNull. And I dont know why
    }
    JsNull
  }
```
Then I ask Benoit to find a work around, that is the current impl or Mathiew's proposal, and It uses the Optional<Long> :
```java
  implicit val quotaValueWrites: Writes[QuotaValue[_]] = value => OptionConverters.toScala(value.asOptional()) match {
    case Some(value) =>  JsNumber(BigDecimal.decimal(value))
    case None => JsNull
  }
```

@chibenwa @mbaechler Could you help me to find a way to correct the first impl of mine?

Reply:
We used to treated ulimited QuotaValue in quota-mailing as null property in JSON string

Reply:
I thought null were for undefined quota, -1 for unlimited.

Reply:
It works! thaks :smile: 
```java
implicit val quotaValueWrites: Writes[QuotaValue[_]] = value => if (value.isUnlimited) JsNull else JsNumber(value.asLong())
```

Reply:
please check it out #2025

Reply:
this is what I saw in quota-mailing-cassandra, `limit` is an Optional<Long> but not a Long. Is it correct? If no, we should open a ticket.
```java
class QuotaDTO {
    @JsonCreator
    private QuotaDTO(@JsonProperty("used") long used,
                     @JsonProperty("limit") Optional<Long> limit) {
        this.used = used;
        this.limit = limit;
    }
}
```

Reply:
seems you are right. What I said is true for limits, but not for usagE.

Reply:
IMO we don't need to be consistent here with the webadmin behaviour

As we don't have to rely with the 'remove the limit case', the scenario is different.

IE the difference of commands (where you ask for a change IE the webadmin case) and event (notification of the status of a system).

Reply:
Agree, the idea of not exposing the internal implementation is exactly to be able to serialize it as you want depending on the context.